pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/buffer/Bytes.java

```diff
@@ -353,4 +353,31 @@ private void validateOffset(long offset) {
             throw new IndexOutOfBoundsException("offset=" + offset + ", length=" + this.length);
         }
     }
+
+    /** {@inheritDoc} */
+    @Override
+    public int compareTo(@NonNull Bytes other) {
+        long lThis = length();
+        long lOther = other.length();
+        if (lThis != lOther) {
+            return lThis > lOther ? 1 : -1;
+        }
+
+        try {
+            InputStream thisBytes = this.toInputStream();
+            InputStream otherBytes = other.toInputStream();
+            for (int i = 0; i < lThis; i++) {
+                int thisVal = thisBytes.read();
+                int otherVal = otherBytes.read();
+                if (thisVal != otherVal) {
+                    return thisVal > otherVal ? 1 : -1;
+                }
+            }
+        }
```

Comment:
I am not sure about the secondary sort order. We should check with @rbair23

Reply:
Yes! I thought comparing bytes will be fine, since it sounds like it is important to get the order right. But, yes asking for the thoughts of @rbair23 is a great idea.

Reply:
I added some comment to the issue https://github.com/hashgraph/pbj/issues/36#issuecomment-1546363015. Reposting here:

>I'm sure there are several interesting ways to sort (in my case, I want length and then by order). Rather than Bytes being comparable itself, we could just have Bytes have some built in comparators. You could have SORT_BY_LENGTH and SORT_BY_VALUE and then combine them Comparator.thenComparing, like SORT_BY_LENGTH.thenComparing(SORT_BY_VALUE). That way we don't have to pick just one way.